python_bindings/dda_model/binarizers.py

```diff
@@ -0,0 +1,66 @@
+import numpy as np
+import math
+import matplotlib.pyplot as plt
+
+def parabolic(parameters):
+    return parameters - np.square(parameters)
+
+def gradParabolic(parameters):
+    return 1 - 2 * parameters
+
+def gaussian(parameters, sig, mu):
+    return 1/(np.sqrt(2*np.pi)*sig)*np.exp(-np.power((parameters - mu)/sig, 2)/2)/24
+
+def gradGaussian(parameters, sig, mu):
+    return -(parameters - mu) * gaussian(parameters, sig, mu) / np.square(sig)
+
+def triangular(parameters, slope):
+    result = np.array(parameters)
+
+    for i, para in enumerate(parameters):
+        if(para <= 0.5):
+            result[i] = slope*para
+        else:
+            result[i] = slope*(1-para)
+
+    return result
+
+def gradTriangular(parameters, slope):
+    result = np.array(parameters)
+
+    for i, para in enumerate(parameters):
+        if(para <= 0.5):
+            result[i] = slope
+        else:
+            result[i] = -slope
+
+    return result
+
+def piecewise_update(x, x_max, y_min, y_max):
+    if x <= 0.5 * x_max:
+        return y_min
+    elif 0.5 * x_max < x <= 0.7 * x_max:
+        return y_min + (y_max - y_min) / 5
+    elif 0.7 * x_max < x <= 0.8 * x_max:
+        return y_min + (y_max - y_min) / 2.5
+    else:
+        return y_max
+    
+def exp_update(x, x_max, y_min, y_max):
```

Comment:
base and the 300 threshold should be arguments to this function with default values of 100 and 300. This will let you set them from the calculation script, which means you can set them in the config.json.

Reply:
done!